lib/methods/pluginnotificationtask.cpp

```diff
@@ -62,8 +62,26 @@ void PluginNotificationTask::ScriptFunc(const Notification::Ptr& notification,
 		callback = [checkable](const Value& commandline, const ProcessResult& pr) { ProcessFinishedHandler(checkable, commandline, pr); };
 	}
 
+	Array::Ptr safeToTruncate = new Array();
+
+	auto hostCr (host->GetLastCheckResult());
+
+	if (hostCr) {
+		safeToTruncate->Add(hostCr->GetOutput());
+	}
+
+	if (service) {
+		auto cr (service->GetLastCheckResult());
+
+		if (cr) {
+			safeToTruncate->Add(cr->GetOutput());
+		}
+	}
```

Comment:
Why don't you just go with the output from the `cr` parameter and be done with it? After all, that given `cr` belongs to either a host or a service, so why do you need both?

Reply:
You could have a `NotificationCommand` that passes both `$host.output$` and `$service.output$`, where both outputs could be excessively long.